sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/splittabledofn/WatermarkEstimators.java

```diff
@@ -37,14 +37,16 @@
     private Instant lastReportedWatermark;
 
     public Manual(Instant watermark) {
-      this.watermark = checkNotNull(watermark, "watermark must not be null.");
-      if (watermark.isBefore(GlobalWindow.TIMESTAMP_MIN_VALUE)
-          || watermark.isAfter(GlobalWindow.TIMESTAMP_MAX_VALUE)) {
-        throw new IllegalArgumentException(
-            String.format(
-                "Provided watermark %s must be within bounds [%s, %s].",
-                watermark, GlobalWindow.TIMESTAMP_MIN_VALUE, GlobalWindow.TIMESTAMP_MAX_VALUE));
+      checkNotNull(watermark, "watermark must not be null.");
+
+      // Making sure that the watermark is within bounds.
```

Comment:
The bounds are wrong, they should be `BoundedWindow.TIMESTAMP_MIN_VALUE` and `BoundedWindow.TIMESTAMP_MAX_VALUE`. This could have been the issue with the timestamp being out of range. If that is the case then I would rather keep the bounds as they were and make setWatermark more restrictive. We should also update the javadoc for setWatermark/currentWatermark (on the interfaces) saying that the value must always be within these bounds.

I think it makes more sense keep the restriction here and also make the restriction within setWatermark as well so the user gets an error when setting it.

Reply:
What's the difference between those values. Seems like GlobalWindow get's these constants from the super class BoundedWindow ?

Reply:
`MIN` is the same but `GlobalWindow.MAX = BoundedWindow.MAX - 1 day` as can be seen here: https://github.com/apache/beam/blob/25b4ebc65953af34a0cfe83ed80feee0bc6df4ad/model/pipeline/src/main/proto/beam_runner_api.proto#L38

Reply:
Right, but we are using the constant from the super class here (TIMESTAMP_MAX_VALUE) not END_OF_GLOBAL_WINDOW.
https://github.com/apache/beam/blob/25b4ebc65953af34a0cfe83ed80feee0bc6df4ad/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/GlobalWindow.java#L39

Also, the error we saw was following which points to min value being off, not max value.
java.lang.IllegalArgumentException: Provided watermark -290308-12-21T19:59:05.224Z must be within bounds [-290308-12-21T19:59:05.225Z, 294247-01-10T04:00:54.775Z].

Reply:
Your right, it would be good to migrate to use BoundedWindow as the import for the static though.

I think it makes sense to make the constructor validate the bounds and have setWatermark ensure that the value is within the range as expected. We can fix the UnboundedSource SDF wrapper to clamp the watermark value that is being reported from UnboundedReader instead.